{
  try {
    return condition.check(context);
  }
 catch (  GenerationFailureException ex) {
    throw ex;
  }
catch (  Throwable t) {
    context.showErrorMessage(null,"IF macro condition failed");
    throw new GenerationFailureException(t);
  }
}
